File: benchmark/fib_iter.c
#include "bench_util.h"

int main(void) {
  const int64_t steps = bench_env_i64("BENCH_FIB_N", 30000000);
  uint64_t a = 0, b = 1;
  int64_t t0 = bench_now_ms();
  for (int64_t i = 0; i < steps; ++i) {
    uint64_t next = a + b;
    a = b;
    b = next;
  }
  int64_t ms = bench_now_ms() - t0;
  bench_print_line("C", "fib_iter", ms, b);
  return 0;
}

File: benchmark/fib_iter.py
#!/usr/bin/env python3
import os
import time

STEPS = int(os.environ.get("BENCH_FIB_N", "30000000"))

def fib_iter() -> int:
    a, b = 0, 1
    mask = (1 << 64) - 1
    for _ in range(STEPS):
        a, b = b, (a + b) & mask
    return b

def main() -> None:
    t0 = time.perf_counter()
    result = fib_iter()
    ms = int((time.perf_counter() - t0) * 1000)
    print(f"{'Python':<8} {'fib_iter':<14} {ms:8d} ms  result={result}")

if __name__ == "__main__":
    main()
